addons/stock/models/stock_picking.py

```diff
@@ -626,6 +626,25 @@ def action_done(self):
         self.write({'date_done': fields.Datetime.now()})
         return True
 
+    @api.depends('state', 'move_lines', 'move_lines.state', 'move_lines.package_level_id')
+    def _compute_move_without_package(self):
+        for picking in self:
+            move_ids_without_package = self.env['stock.move']
+            if picking.picking_type_entire_packs == False:
+                move_ids_without_package = picking.move_lines
+            else:
+                for move in picking.move_lines:
+                    if not move.package_level_id:
+                        if move.state in ('assigned', 'done'):
+                            if any(not ml.package_level_id for ml in move.move_line_ids):
```

Comment:
if not all(ml.package_level_id for ml in move.move_line_ids) seems more readable. your call